### ISSUE TITLE
HTTP request lost

### ISSUE THREAD

**Issue by btoonk** [istio/istio#23957]
**Bug description**

An app with Istio enabled connects to a http service outside the mesh during startup.
Most (or all initial) request fails. In the istio-proxy container log we don't see the request getting logged. With debug enabled we do see the following:

```
[Envoy (Epoch 0)] [2020-05-18 19:24:09.307][38][debug][filter] [external/envoy/source/extensions/filters/listener/original_dst/original_dst.cc:18] original_dst: New connection accepted
[Envoy (Epoch 0)] [2020-05-18 19:24:09.308][38][debug][filter] [external/envoy/source/extensions/filters/listener/tls_inspector/tls_inspector.cc:78] tls inspector: new connection accepted
[Envoy (Epoch 0)] [2020-05-18 19:24:09.408][38][debug][conn_handler] [external/envoy/source/server/connection_handler_impl.cc:196] listener filter times out after 100 ms
[Envoy (Epoch 0)] [2020-05-18 19:24:09.408][38][debug][conn_handler] [external/envoy/source/server/connection_handler_impl.cc:199] fallback to default listener filter
[Envoy (Epoch 0)] [2020-05-18 19:24:09.408][38][debug][conn_handler] [external/envoy/source/server/connection_handler_impl.cc:330] closing connection: no matching filter chain found
[Envoy (Epoch 0)] [2020-05-18 19:24:09.548][37][debug][connection] [external/envoy/source/common/network/connection_impl.cc:538] [C138] remote close
[Envoy (Epoch 0)] [2020-05-18 19:24:09.548][37][debug][connection] [external/envoy/source/common/network/connection_impl.cc:192] [C138] closing socket: 0
[Envoy (Epoch 0)] [2020-05-18 19:24:09.548][37][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:124] [C138] client disconnected
[Envoy (Epoch 0)] [2020-05-18 19:24:09.548][37][debug][connection] [external/envoy/source/common/network/connection_impl.cc:101] [C137] closing data_to_write=31 type=0
[Envoy (Epoch 0)] [2020-05-18 19:24:09.548][37][debug][pool] [external/envoy/source/common/tcp/conn_pool.cc:238] [C138] connection destroyed
```

We're using the scuttle wrapper for this app so the istio-proxy is already running once the app starts.
The service that it's connecting to is running on port 8080 (http).
Looking at the listener on 0.0.0.0:8080 I don't see any reference to the PassthroughCluster cluster in the filters.

Due to an other issue we had with an external httpS service on port 8080 we created a ServiceEntry with the following spec (this SE has nothing todo with the svc the app connects to):

```
  hosts:
  - host1.domain.com
  location: MESH_EXTERNAL
  ports:
  - name: https-8080
    number: 8080
    protocol: HTTPS
  resolution: DNS
```

Once we add another host (any dummy host) to this serviceEntry the listener on 0.0.0.0:8080 changes and does includes the PassthroughCluster cluster:

```
                    {
                        "name": "envoy.tcp_proxy",
                        "typedConfig": {
                            "@type": "type.googleapis.com/envoy.config.filter.network.tcp_proxy.v2.TcpProxy",
                            "statPrefix": "PassthroughCluster",
                            "cluster": "PassthroughCluster",
                            "accessLog": [
                                {
                                    "name": "envoy.file_access_log",
...
                                    }
                                }
                            ]
                        }
                    }
```

With this change the app is able to startup and connect to the service outside the mesh without any problems. Also the request appears in the access logging in envoy.

To me, this is like a bug in istiod where building the filters doesn't always go well.

**Expected behavior**

The service should be able to do a http request to the external mesh service.

**Steps to reproduce the bug**

I'm not sure how to reproduce, we currently have only a single app that is having issues.
It looks like adding the serviceEntry does influence this.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

We tested with Istio 1.5.1 and 1.5.2

**How was Istio installed?**
istioctl manifest apply

**Environment where bug was observed (cloud vendor, OS, etc)**
GKE v1.15.9-gke.26

**Comment by rolandkool** [comment]
The fix in 1.6 has indeed resolved the issue for us.

**Comment by howardjohn** [comment]
Thanks!
